sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -65,18 +72,16 @@ trait HashJoin {
   {
     new Iterator[InternalRow] {
       private[this] var currentStreamedRow: InternalRow = _
-      private[this] var currentHashMatches: CompactBuffer[InternalRow] = _
+      private[this] var currentHashMatches: Seq[InternalRow] = _
```

Comment:
is there any reason why you are changing it to a more generic type? it's usually better to use a tighter type because JIT can optimize it better.